sdk/core/azure-json/src/main/java/com/azure/json/JsonArray.java

```diff
@@ -0,0 +1,377 @@
+     * object to return.
+     * @return the JsonElement by index from the JsonArray
+     * @throws IndexOutOfBoundsException Thrown when index parameter is < 0 or
+     * >= size()
+     */
+    public JsonElement getElement(int index) throws IndexOutOfBoundsException {
+        return this.elements.get(index);
+    }
+
+    /**
+     * Removes a specified element according to its index from the JsonArray,
+     * shifting the existing elements from that index to the left and decrementing
+     * the size of the JsonArray by 1.
+     *
+     * @param index the index specifying which JsonElement from the JsonArray
+     * object to remove.
+     * @return the resulting JsonArray after the specified JsonElement has been
+     * removed.
+     * @throws IndexOutOfBoundsException Thrown when index parameter is < 0 or
+     * >= size()
+     */
+    public JsonElement removeElement(int index) throws IndexOutOfBoundsException {
+        return this.elements.remove(index);
+    }
+
+    public int arrayLength(){
+        return elements.size();
+    }
+
+    /**
+     * Returns the String representation of the JsonArray object
+     * utilising the JsonWriter to abstract. It passes a StringWriter to the
+     * toWriter method and then returns the resulting String.
+     * @return String representation of the JsonArray object
+     */
+    public String toJson() throws IOException {
+        try (StringWriter stringOutput = new StringWriter()) {
+            toWriter(stringOutput);
+            return stringOutput.toString();
+        }
+    }
+
+    /**
+     * Takes a writer and uses it to serialize the JsonArray object
+     * @param writer the writer to use, which is then wrapped in a JsonWriter
+     * @return the writer
+     * @throws IOException if the writer throws an exception
+     */
+    public Writer toWriter(Writer writer) throws IOException {
+        try (JsonWriter jsonWriter = JsonProviders.createWriter(writer)) {
+            serialize(jsonWriter);
+        }
+        return writer;
+    }
+
+    /**
+     * Takes an output stream and uses it to serialize the JsonArray object
+     * @param stream the output stream to use, which is then wrapped in a JsonWriter
+     * @return the output stream
+     * @throws IOException if the output stream throws an exception
+     */
+    public OutputStream toStream(OutputStream stream) throws IOException {
+        try (JsonWriter jsonWriter = JsonProviders.createWriter(stream)) {
+            serialize(jsonWriter);
+        }
+        return stream;
+    }
+
+    /**
+     * Takes a JsonWriter and uses it to serialize the JsonArray object. At
+     * every step, it utilises serialize() method, which is inherited from the
+     * parent. This means essentially it doesn't matter what JsonElement is
+     * passed in to the add method, all it has to do is pass the JsonWriter to
+     * the serialize() method of whatever JsonElement it is using, and it will
+     * serialize.
```

Comment:
Too much implementation detail in the JavaDoc, which should just concern itself with what the end user of the API is needing to know.

Reply:
@JonathanGiles  have rewritten